flow/workflows/cancel_table_addition_flow.go

```diff
@@ -0,0 +1,284 @@
+	// Start new CDC flow as a regular workflow via activity
+	createFlowOptions := workflow.ActivityOptions{
+		StartToCloseTimeout: time.Minute * 5,
+	}
+	createFlowCtx := workflow.WithActivityOptions(ctx, createFlowOptions)
+
+	err = workflow.ExecuteActivity(createFlowCtx, cancelTableAddition.StartNewCDCFlow,
+		flowConfig, state, originalWorkflowId).Get(ctx, nil)
+	if err != nil {
+		logger.Error("Failed to create job entry and start new CDC flow", "error", err)
+		return nil, err
+	}
+
+	logger.Info("Successfully started CDC flow with updated configuration",
+		"flowName", flowJobName,
+		"workflowId", originalWorkflowId,
+		"originalTableCount", len(flowConfig.TableMappings),
+		"finalTableCount", len(finalListOfTables),
+		"completedTableCount", len(snapshottedSourceTables),
+		"oidMappingCount", len(tableOIDs))
+
+	waitForRunningMirrorOptions := workflow.ActivityOptions{
+		StartToCloseTimeout: time.Minute * 10,
+		RetryPolicy: &temporal.RetryPolicy{
+			InitialInterval:    time.Second * 10,
+			BackoffCoefficient: 2.0,
+			MaximumInterval:    time.Minute * 2,
+			MaximumAttempts:    10,
+		},
+	}
+	waitForRunningMirrorCtx := workflow.WithActivityOptions(ctx, waitForRunningMirrorOptions)
+
+	err = workflow.ExecuteActivity(
+		waitForRunningMirrorCtx,
+		cancelTableAddition.WaitForNewRunningMirrorToBeInRunningState,
+		flowJobName,
+		originalWorkflowId,
+	).Get(ctx, nil)
+	if err != nil {
+		logger.Error("Failed to confirm new mirror is running", "error", err)
+		return nil, err
+	}
+
+	logger.Info("CDC workflow started and running",
+		"flowName", flowJobName,
+		"finalTableCount", len(finalListOfTables))
+
+	removeCancelledTablesFromPublicationOptions := workflow.ActivityOptions{
+		StartToCloseTimeout: 10 * time.Minute,
+		HeartbeatTimeout:    2 * time.Minute,
+		RetryPolicy: &temporal.RetryPolicy{
+			InitialInterval:    time.Second * 10,
+			BackoffCoefficient: 2.0,
+			MaximumInterval:    time.Minute * 5,
+		},
+	}
+	removeCancelledTablesFromPublicationCtx := workflow.WithActivityOptions(ctx, removeCancelledTablesFromPublicationOptions)
+	err = workflow.ExecuteActivity(
+		removeCancelledTablesFromPublicationCtx,
+		cancelTableAddition.RemoveCancelledTablesFromPublicationIfApplicable,
+		flowJobName,
+		flowConfig.SourceName,
+		flowConfig.PublicationName,
+		finalListOfTables,
+	).Get(ctx, nil)
+	if err != nil {
+		logger.Error("Failed to remove cancelled tables from publication", "error", err)
+		return nil, err
+	}
+
+	cleanupIncompleteTablesOptions := workflow.ActivityOptions{
+		StartToCloseTimeout: time.Minute * 15,
+		HeartbeatTimeout:    2 * time.Minute,
+		RetryPolicy: &temporal.RetryPolicy{
+			InitialInterval:    time.Second * 10,
+			BackoffCoefficient: 2.0,
+			MaximumInterval:    time.Minute * 5,
+		},
+	}
+	cleanupIncompleteTablesCtx := workflow.WithActivityOptions(ctx, cleanupIncompleteTablesOptions)
+
+	err = workflow.ExecuteActivity(cleanupIncompleteTablesCtx, cancelTableAddition.CleanupIncompleteTablesInStats,
```

Comment:
Just remembered, there's also `CleanupQRepFlow` to drop the stage. I think we can return the deleted flow_names from table cleanup and change the args of `dstConn.CleanupQRepFlow` to `stagingPath, flowJobName string` and should be good to do an activity